src/components/Dialogs/ActionsModal.tsx

```diff
@@ -87,31 +83,44 @@ type Props = {
   onClose: () => void;
 };
 
-const Action = ({ actionKey }: { actionKey: string }) => {
+const Action = ({ actionKey, length }: { actionKey: string, length: number }) => {
   const { mutate: runAction } = useRunActionMutation();
+  const [hover, setHover] = useState(false);
 
   const action = useMemo(() => quickActions[actionKey], [actionKey]);
   const { functionName, name } = action;
 
   const handleAction = useEventCallback(() => {
+    setHover(true);
     runAction(functionName, {
       onSuccess: () => toast.success(`Running action "${name}"`),
     });
+    setTimeout(() => {
+      setHover(false);
+    }, 2000);
   });
 
+  const handleMouseEnter = () => {
+    setHover(true);
+  };
+  const handleMouseLeave = () => {
+    setHover(false);
+  };
+
   return (
-    <TransitionDiv className="mr-4 flex flex-row justify-between gap-y-2 border-b border-panel-border pb-4 last:border-0">
-      <div className="flex w-full max-w-[35rem] flex-col gap-y-2">
-        <div>{name}</div>
-        <div className="text-sm opacity-65">{quickActions[actionKey].info}</div>
+    <div
+      className={cx('flex flex-row justify-between gap-y-2 cursor-pointer', length > 5 ? 'mr-4' : '')}
+      onClick={handleAction}
+      onMouseEnter={handleMouseEnter}
+      onMouseLeave={handleMouseLeave}
+    >
+      <div className="flex w-full flex-col gap-y-1">
```

Comment:
Is it not easier to put `hover:text-panel-text-primary` here and remove all the `setHover`?

Reply:
I tried, this way the hover works even if you come from the bottom and hit the description first.